kayenta-core/src/main/java/com/netflix/kayenta/security/AccountCredentials.java

```diff
@@ -17,39 +17,44 @@
 package com.netflix.kayenta.security;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
 
-public interface AccountCredentials<T> {
-  String getName();
+@Getter
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(of = "name")
+public abstract class AccountCredentials<T> {
```

Comment:
Good thing about making this abstract is that it'll fit more naturally as an implementation of `Credentials` from Kork. Considering that I've been consolidating a lot of account management stuff into Kork lately, this will be good.